src/Gui/SoDatumLabel.cpp

```diff
@@ -451,6 +522,7 @@ SbVec3f SoDatumLabel::getLabelTextCenter()
     const SbVec3f* points = this->pnts.getValues(0);
     SbVec3f p1 = points[0];
     SbVec3f p2 = points[1];
+    SbVec3f p3 = points[2];
```

Comment:
When building FreeCAD with the address sanitizer option I get a reliable crash here. Apparently the points field doesn't have four points.

Steps to reproduce:
* Create a sketch
* Try to create a rectangle
* It crashes when clicking inside the 3D view

Reply:
`points` has 3 fields only in `ARCLENGTH` case, it has 1 field in `ANGLE` one and 2 in others. A check can be added.

Reply:
https://github.com/FreeCAD/FreeCAD/pull/13627